trestle/oscal/assessment_results.py

```diff
@@ -1707,6 +1717,43 @@ class Action(OscalBaseModel):
     remarks: Optional[Remarks] = None
 
 
+class Activity(OscalBaseModel):
+    uuid: constr(
+        regex=r'^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}$'
+    ) = Field(
+        ...,
+        description='Uniquely identifies this assessment activity. This UUID may be referenced elsewhere in an OSCAL document when refering to this information. A UUID should be consistantly used for a given included activity across revisions of the document.',
+        title='Assessment Activity Universally Unique Identifier',
+    )
+    title: Optional[str] = Field(
+        None,
+        description='The title for this included activity.',
+        title='Included Activity Title',
+    )
+    description: str = Field(
+        ...,
+        description='A human-readable description of this included activity.',
+        title='Included Activity Description',
+    )
+    props: Optional[List[Property]] = Field(None, min_items=1)
+    annotations: Optional[List[Annotation]] = Field(None, min_items=1)
+    links: Optional[List[Link]] = Field(None, min_items=1)
+    actions: Optional[List[Action]] = Field(None, min_items=1)
+    related_controls: Optional[ReviewedControls] = Field(None, alias='related-controls')
+    responsible_roles: Optional[Dict[str, ResponsibleRole]] = Field(
+        None, alias='responsible-roles'
+    )
+    remarks: Optional[Remarks] = None
+
+
+class LocalDefinitions(OscalBaseModel):
+    add_objectives_and_methods: Optional[List[LocalObjective]] = Field(
+        None, alias='add-objectives-and-methods', min_items=1
+    )
+    activities: Optional[List[Activity]] = Field(None, min_items=1)
+    remarks: Optional[Remarks] = None
```

Comment:
Should there not be inventory_items here?

Reply:
As discussed, the oscal schema has local-definitions defined locally twice in separate locations, and they have different contents.  The fix is to allow the creation of two separate classes, LocalDefinitions and LocalDefinitions1 - as created by datamodel-codegen.  The correct one should be used with the corresponding parent class - and pydantic should enforce that agreement.  The fundamental problem appears to be in the oscal schema itself.  This happens in many places so users must be aware of the possibility for multiple classes differing only by the index.